hostapd: Enable DPP support in full TLS variant

gubertoli opened the pull request:
Enable Device Provisioning Protocol* (DPPv2) in hostapd, but only for full 
builds that use external TLS libraries.
\* aka Wi-Fi Easy Connect

Signed-off-by: Gustavo Bertoli <gubertoli@gmail.com>

gubertoli committed:
hostapd: Enable DPPv2 support in full TLS variant

Enable Device Provisioning Protocol
(DPPv2) in hostapd, but only for full
builds that use external TLS libraries
(OpenSSL or WolfSSL).

Signed-off-by: Gustavo Bertoli <gubertoli@gmail.com>

gubertoli commented:
@robimarko thank you for the clarification and guidance in #18232.

I have now created a clean version of the PR. Only enabling DPPv2 for 
full TLS variants of OpenSSL and WolfSSL.

robimarko commented on package/network/services/hostapd/Makefile:
Is mbedtls not enough?

gubertoli replied:
By https://github.com/openwrt/openwrt/issues/10303#issuecomment-1200318510, seems that mbedtls is not working with DPP.

robimarko replied:
That is highly outdated, please test it with mbedtls, it should really work

gubertoli replied:
I've updated the hostapd Makefile to enable DPP for wpad-openssl and wpad-mbedtls, and dropped it from wpad-wolfssl since it fails to build due to missing PKCS7 and certificate support in the default WolfSSL config.

gubertoli commented on package/network/services/hostapd/files/hostapd-full.config:
The idea was to keep aligned with [hostapd.conf](https://w1.fi/cgit/hostap/tree/hostapd/defconfig#n418) from hostap.

systemcrash replied:
I don't think it's a bad idea to keep some of these around. Might be confusing tho.

gubertoli committed:
hostapd: enable DPP in wpad-mbedtls, drop from wpad-wolfssl

The wpad-mbedtls variant now explicitly enables CONFIG_DPP and
CONFIG_DPP2.

Drop DPP support from wpad-wolfssl as it currently fails to build due
to missing PKCS7 and certificate handling features in the default
OpenWrt WolfSSL configuration.

Signed-off-by: Gustavo Bertoli <gubertoli@gmail.com>

systemcrash commented:
Looks OK.

systemcrash commented:
Please don't include merge commits - rebase on `main` (or `master`)

blocktrron commented:
Squashed ans applied with some changes to my staging tree - thanks!
